core/src/main/kotlin/knote/api/Page.kt

```diff
@@ -1,21 +1,27 @@
 package knote.api
 
 import knote.script.PageScript
+import knote.util.KObservableList
+import knote.util.KObservableMap
 import knote.util.KObservableObject
 import java.io.File
+import java.nio.file.Path
 import kotlin.script.experimental.api.ScriptDiagnostic
 
 interface Page {
     val id: String
     val fileObject: KObservableObject<Page, File>
     val fileContentObject: KObservableObject<Page, String>
+    val textObject: KObservableObject<Page, String>
     val compiledScriptObject: KObservableObject<Page, PageScript?>
     val reportsObject: KObservableObject<Page, List<ScriptDiagnostic>?>
     val resultObject: KObservableObject<Page, Any?>
     val dependenciesObject: KObservableObject<Page, Set<String>>
+    val fileInputs: KObservableList<Path>
 
     val file get() = fileObject.value
     val fileContent get() = fileContentObject.value
+    val text get() = textObject.value
     val compiledScript get() = compiledScriptObject.value
     val reports get() = reportsObject.value
     val result get() = resultObject.value
```

Comment:
for increased readability I usually see `gets` done something along the lines of 
```
interface Page {
	...
	
	val file
		get() = fileObject.value
	val fileContent
		get() = fileContentObject.value
```

BUT NOW I'm think about it a little more... maybe we could do this instead?
for increased readability I usually see `gets` done something along the lines of 
```
interface Page {

	val id: String
	val fileObject: KObservableObject<Page, File>
			get() = fileObject.value
	val fileContentObject: KObservableObject<Page, String>
			get() = this.value // I can't remember which it is in this case
	val textObject: KObservableObject<Page, String>
	...
```

Well idk I now remember that we wanted to be able to differentiate getting both types... I'm going to research this a little